assets/js/data/utils/update-state.ts

```diff
@@ -1,33 +1,37 @@
 /**
- * External dependencies
+ * Utility for updating nested state in the path that changed.
  */
-import { klona } from 'klona/json';
+function updateNested< Type >( // The state being updated
+	state: Type,
+	// The path being updated
+	path: string[],
+	// The value to update for the path
+	value: unknown,
+	// The current index in the path
+	index: number
```

Comment:
It would be cool if this defaulted to `0`, that way the consumer doesn't have to specify that.